logic/src/commonMain/kotlin/com/wire/kalium/logic/network/SessionManagerImpl.kt

```diff
@@ -49,29 +69,59 @@ class SessionManagerImpl internal constructor(
         serverConfig.get()!!
     }
 
-    override fun updateLoginSession(newAccessTokeDTO: AccessTokenDTO, newRefreshTokenDTO: RefreshTokenDTO?): SessionDTO =
+    override suspend fun updateLoginSession(newAccessTokenDTO: AccessTokenDTO, newRefreshTokenDTO: RefreshTokenDTO?): SessionDTO? =
         wrapStorageRequest {
             tokenStorage.updateToken(
                 userId = idMapper.toDaoModel(userId),
-                accessToken = newAccessTokeDTO.value,
-                tokenType = newAccessTokeDTO.tokenType,
+                accessToken = newAccessTokenDTO.value,
+                tokenType = newAccessTokenDTO.tokenType,
                 refreshToken = newRefreshTokenDTO?.value
             )
         }.map {
             sessionMapper.fromEntityToSessionDTO(it)
         }.onSuccess {
             session.set(it)
-        }.fold({
-            TODO("IMPORTANT! Not yet implemented")
+        }.nullableFold({
+            null
         }, {
             it
         })
 
-    override suspend fun onSessionExpired() {
+    override suspend fun updateToken(accessTokenApi: AccessTokenApi, oldAccessToken: String, oldRefreshToken: String): SessionDTO? {
```

Comment:
Future refactoring: inject `AccessTokenApi` into `SessionManagerImpl` and we can remove it has a parameter and we don't need to forward it all the way down to the HttpClient.